src/pages/LandingPage.js

```diff
@@ -1,5 +1,123 @@
-import React from "react";
+import React, { useState } from "react";
+import {
+  Button,
+  CircularProgress,
+  Grid,
+  TextField,
+  Typography,
+} from "@material-ui/core";
+import { useForm, Controller } from "react-hook-form";
+import { makeStyles } from "@material-ui/core/styles";
+import { useHistory } from "react-router-dom";
+import { StatusBar } from "../components/StatusBar";
+
+const useStyles = makeStyles(() => ({
+  root: {
+    flexGrow: 1,
+  },
+  formContainer: {
+    minHeight: "80vh",
+  },
+  fullWidthContainer: {
+    width: "100%",
+  },
+  welcomeText: {
+    textAlign: "center",
+  },
+}));
 
 export const LandingPage = () => {
-  return <div>LandingPage</div>;
+  const classes = useStyles();
+  const {
+    control,
+    handleSubmit,
+    errors,
+    formState: { isSubmitting },
+  } = useForm();
+  const history = useHistory();
+  const [openStatus, setOpenStatus] = useState(false);
+  const [statusMessage, setStatusMessage] = useState("");
+
+  const onSubmit = async (data) => {
+    //Just a delay when testing
+    const response = await new Promise((resolve) => {
+      setTimeout(() => resolve(false), 1000);
+    });
+
+    console.log(data);
+    if (response) {
+      history.push(`/vote/${data.pin}`);
+    } else {
+      setStatusMessage(!!response.text ? response.text : "");
+      setOpenStatus(true);
+    }
+  };
+
+  return (
+    <div className={classes.root}>
+      <StatusBar open={openStatus} setOpen={setOpenStatus} severity="error">
+        {statusMessage.length !== 0
+          ? statusMessage
+          : "Something went wrong, could not find poll"}
+      </StatusBar>
+      <Grid
+        container
+        justify="center"
+        alignItems="center"
+        className={classes.formContainer}
+        direction="column"
+        spacing={2}
+      >
+        <Grid item xs={12} sm={6}>
+          <Typography variant="subtitle1" className={classes.welcomeText}>
+            Welcome to Survey Service! To vote enter a pin below, or login in to
+            create your own poll!
+          </Typography>
+        </Grid>
+        <Grid container item xs={10} sm={4} md={3} justify="center">
+          <form
+            onSubmit={handleSubmit(onSubmit)}
+            className={classes.fullWidthContainer}
+          >
+            <Grid item>
+              <Controller
+                as={TextField}
+                rules={{
+                  required: "Pin code is required",
+                  pattern: {
+                    value: /^[a-zA-Z0-9]{6}$/,
+                    message: "Must contain 6 chars",
```

Comment:
Kanskje vi bør endre "Must contain 6 chars" til "Pin must be of length 6, can be both chars and digits" eller noe sånn, siden ellers kan det virke som at det bare er bokstaver man kan skrive inn? 👍

Reply:
Ja, godt poeng! Gir ikke brukeren så mye info.